tcpserver.c

```diff
@@ -163,3 +165,44 @@ void close_server(void)
     int status = close(own_socket);
     printf("Attempted closing server with status: %d\n", status);
 }
+
+/**
+* Add a new element to client_sockets
+* @param new_element element to be appended to client_sockets
+*/
+void add_to_array(int new_element) 
+{
+    int *new_array = reallocarray(client_sockets, n_connected_clients + 1, sizeof(int));
+    new_array[n_connected_clients++] = new_element;
+    client_sockets = new_array;
+}
+
+/**
+* Remove del_element from client_sockets
+* @param del_element element to be deleted from client_sockets
+*/
+void rm_from_array(int del_element)
+{
+    int *new_array = calloc(n_connected_clients - 1, sizeof(int));
+    int idx = 0;
+    for (int i = 0; i < n_connected_clients; i++)
+    {
+        if (client_sockets[i] != del_element)
```

Comment:
Suggest that you only remove the element requested to be deleted and maybe replaces it with -1 or some other scheme.